ephemeris_tools_unit_tests/ephem_tools_unit_tests.py

```diff
@@ -0,0 +1,702 @@
+            viewer_replace(ephem, tests_file, golden_copies_location)
+        if 'tracker3' in test_urls[0]:
+            moon_tracker_replace(ephem, tests_file, golden_copies_location)
+        if 'ephem3' in test_urls[0]:
+            ephemeris_generator_replace(ephem, tests_file, 
+                                        golden_copies_location)
+            
+current_time = datetime.now().strftime('%H:%M:%S')
+logging.basicConfig(filename=f'ephem_tools_unit_test_{current_time}.log',
+                    encoding='utf-8',
+                    level=logging.DEBUG, 
+                    format='%(asctime)s | %(levelname)s | %(message)s',
+                    datefmt='%y-%m-%d %H:%M:%S',
+                    force=True)
+    
+warnings.simplefilter('ignore', urllib3.exceptions.InsecureRequestWarning)
+logging.getLogger('urllib3').setLevel(logging.WARNING)
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--run-test-ephemeris', dest='chosenephems', 
+                    action='append_const', const='test',
+                    help='Runs your chosen action with the "test" '
+                         'extension implemented to force all tests to operate '
+                         'with the most recent ephemeris.')
+
+parser.add_argument('--run-current-ephemeris', dest='chosenephems', 
+                    action='append_const', const='current',
+                    help='Runs your chosen action in the current active '
+                         'version of the server.')
+
+parser.add_argument('tool_directory', type=str, nargs=1,
+                    help='The location of the URL source files that will '
+                         'generate the results. All files called must be '
+                         'within the same location within the directory. '
+                         'Do not include the file name within the path. '
+                         'Do not end the path with a /.')
+
+parser.add_argument('golden_directory', type=str, nargs=1, 
+                    help='Path to the golden_files storage.')
+
+parser.add_argument('--comparison', action='store_true',
+                    help='Compares results of chosen ephemeris tools to '
+                         'their stored golden version counterparts. '
+                         'The generated results are pulled from your '
+                         'chosen server.')
+
+parser.add_argument('--replacement', dest='folderaction', action='store_false',
+                    help='Replaces stored versions of chosen ephemeris '
+                         'tools. All versions stored are generated from the '
+                         'current staging server.')
+
+parser.set_defaults(folderaction=True)
+
+parser.add_argument('--tools', type=str, nargs='+',
+                    help='The ephemeris tool file(s) you wish to utilize. '
+                         'Any combination of tools is allowed.')
+
+parser.add_argument('--chosen-tests', dest='testsubset', action='store', nargs='+',
```

Comment:
It's generally a bad idea to mix `nargs` with position arguments (`golden_directory` and `tool_directory`). For example, your current code permits this:

```
> python ephem_tools_unit_tests.py tool_directory golden_directory --chosen-tests 1 2 3
```

but not this:

```
> python ephem_tools_unit_tests.py --chosen-tests 1 2 3 tool_directory golden_directory --chosen-tests 1 2 3
usage: ephem_tools_unit_tests.py [-h] [--run-test-ephemeris] [--run-current-ephemeris] [--comparison] [--replacement]
                                 [--tools TOOLS [TOOLS ...]] [--chosen-tests TESTSUBSET [TESTSUBSET ...]]
                                 [--server {staging,production,other}]
                                 tool_directory golden_directory
ephem_tools_unit_tests.py: error: the following arguments are required: tool_directory, golden_directory
```

Instead make it take a single comma-separated string like `1,2,3` and then use `split` to break the arguments apart.

Reply:
Also I think `--limit-to-tests` or something similar would make more sense.